### PR TITLE
feat: Add JSON schema for egg.json

### PR DESCRIPTION
**Pull request by ebebbington**
Huge thanks to @qu4k, who helped me understand these types of schemas, which eventually led to me being able to implement this feature

**Example Usage**

```
// egg.json
{
  $schema: "https://deno.land/x/eggs@<version>/schema.json

  ... // write config as usual
}
```

**Screenshots of Feature in Action**

image

image

image

image

### CODE REVIEW

File: schema.json [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      "description": "All the files that should be ignored when uploading to nest.land. Supports file globbing.",
+      "examples": [["tests/**/*"]]
+    },
+    "checkFormat": {
+      "type": "boolean",
+      "description": "Automatically format your code before publishing to the blockchain. Defaults to false",
+      "default": false
+    },
+    "checkTest": {
+      "type": "boolean",
```
Comment (SteelAlloy) [inline schema.json]: In the last version the type has changed 😉
```suggestion
      "type": ["boolean", "string"],
```

File: schema.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      "description": "All the files that should be uploaded to nest.land. Supports file globbing. Do not use ./**/* for the files field! This has been known to cause errors in the publishing process.",
+      "examples": [["src/**/*"]]
+    },
+    "ignore": {
+      "type": "array",
+      "description": "All the files that should be ignored when uploading to nest.land. Supports file globbing.",
+      "examples": [["tests/**/*"]]
+    },
+    "checkFormat": {
+      "type": "boolean",
```
Comment (SteelAlloy) [inline schema.json]:
```suggestion
      "type": ["boolean", "string"],
```

File: schema.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      "description": "Should people be able to find this module/version on the gallery?. Defaults to false.",
+      "default": false
+    },
+    "repository": {
+      "type": "string",
+      "description": "A link to your repository. Defaults to null.",
+      "default": "null"
+    },
+    "files": {
+      "type": "array",
```
Comment (SteelAlloy) [inline schema.json:46]: Without this the array can contain any type 😉
```suggestion
      "type": "array",
      "items": {
        "type": "string"
      },
```

File: schema.json [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      "description": "All the files that should be uploaded to nest.land. Supports file globbing. Do not use ./**/* for the files field! This has been known to cause errors in the publishing process.",
+      "examples": [["src/**/*"]]
+    },
+    "ignore": {
+      "type": "array",
```
Comment (SteelAlloy) [inline schema.json:54]:
```suggestion
      "type": "array",
      "items": {
        "type": "string"
      },
```

**Review by SteelAlloy — CHANGES_REQUESTED** [review]
Thanks for this addition! 🎉
A few changes to make and everything will be good!

File: schema.json (hunk 1 above)
Comment (ebebbington) [reply schema.json]: Addressed here https://github.com/nestdotland/eggs/pull/87/commits/21651aaaac08f4a443cfc5168ef5732cf526a456 👍

File: schema.json (hunk 2 above)
Comment (ebebbington) [reply schema.json]: Addressed here https://github.com/nestdotland/eggs/pull/87/commits/21651aaaac08f4a443cfc5168ef5732cf526a456 👍

File: schema.json (hunk 3 above)
Comment (ebebbington) [reply schema.json:46]: Addressed here https://github.com/nestdotland/eggs/pull/87/commits/21651aaaac08f4a443cfc5168ef5732cf526a456 👍 (i didn't think about it allowing any type :o)

File: schema.json (hunk 4 above)
Comment (ebebbington) [reply schema.json:54]: Addressed here https://github.com/nestdotland/eggs/pull/87/commits/21651aaaac08f4a443cfc5168ef5732cf526a456 👍

**Review by SteelAlloy — APPROVED** [review]
LGTM ✨

**Pull request merged**